src/App/Pages/Vault/AddEditPage.xaml.cs

```diff
@@ -243,6 +245,12 @@ private async void ScanTotp_Clicked(object sender, System.EventArgs e)
         {
             if (DoOnce())
             {
+                var cameraPermission = await PermissionManager.CheckAndRequestPermissionAsync(new Permissions.Camera());
```

Comment:
Traditionally you'd want the page/activity that makes use of the permission in question to handle the permission check so you don't have to worry about multiple points of entry.  If we're certain this is the only way into ScanPage this will work for now, but we should consider fixing ScanPage at some point in the future (potentially a larger task as we'll need to safely block camera initialization while waiting for permission, etc.)

Reply:
@mportune-bw - This is the only place where I see the explicit instantiation of / navigation to ScanPage. Willing to implement it the "correct way" though if you reach out, I just need a little guidance.

Reply:
Ok cool, let's just roll with it for now then.  Better to stop the crashing sooner 👍